src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1311,8 +1311,183 @@ public static Matrix4x4 CreateReflection(Plane value)
         /// <param name="matrix">The source matrix to invert.</param>
         /// <param name="result">If successful, contains the inverted matrix.</param>
         /// <returns>True if the source matrix could be inverted; False otherwise.</returns>
-        public static bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
+        public static unsafe bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
         {
+            if (Sse.IsSupported)
+            {
+                // This implementation is based on the DirectX Math Library XMMInverse method
+                // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathMatrix.inl
+
+                // Load the matrix values into rows
+                Vector128<float> row1 = Sse.LoadVector128(&matrix.M11);
+                Vector128<float> row2 = Sse.LoadVector128(&matrix.M21);
+                Vector128<float> row3 = Sse.LoadVector128(&matrix.M31);
+                Vector128<float> row4 = Sse.LoadVector128(&matrix.M41);
+
+                // Transpose the matrix
+                Vector128<float> vTemp1 = Sse.Shuffle(row1, row2, 0x44);
+                Vector128<float> vTemp3 = Sse.Shuffle(row1, row2, 0xEE);
+                Vector128<float> vTemp2 = Sse.Shuffle(row3, row4, 0x44);
+                Vector128<float> vTemp4 = Sse.Shuffle(row3, row4, 0xEE);
+
+                row1 = Sse.Shuffle(vTemp1, vTemp2, 0x88);
+                row2 = Sse.Shuffle(vTemp1, vTemp2, 0xDD);
+                row3 = Sse.Shuffle(vTemp3, vTemp4, 0x88);
+                row4 = Sse.Shuffle(vTemp3, vTemp4, 0xDD);
+
+                Vector128<float> V00 = Sse.Shuffle(row3, row3, 0x50);
```

Comment:
The DXMath implementation defines a `Permute` helper function that uses `Avx.Permute` or `Sse.Shuffle` for when `op1`/`op2` are the same.

We could trivially define the following and get the same benefit:
```csharp
[MethodImpl(MethodImplOptions.AggressiveInlining)]
static Vector128<float> Permute(Vector128<float> value, byte control)
{
    if (Avx.IsSupported)
    {
        return Avx.Permute(value, control);
    }

    Debug.Assert(Sse.IsSupported);
    return Sse.Shuffle(value, value, control);
}
```

Reply:
Added back the Permute function using both Avx.Permute and Sse.Shuffle